### PR TITLE
make RMQ connection protocol configurable

### PR DESCRIPTION
**Pull request by aniketdivekar**

### CODE REVIEW

File: src/distributor/rmq_healthcheck_service.ts [hunk 1]
```diff
@@ -93,7 +93,14 @@ export default class RMQModeHeathCheck {
 
   private async getQueues(exchange: string): Promise<string[]> {
     const queues = []
-    const url = `http://${process.env.RMQ_HOST}:15672/api/exchanges/%2F/${exchange}/bindings/source`
+
+    const protocol = process.env.RMQ_PROTOCOL === 'amqps' ? 'https' : 'http'
+    let host = `${process.env.RMQ_HOST}`
+    if (protocol === 'http') {
+      host += ':15672'
```
Comment (devendra-shardeum) [inline src/distributor/rmq_healthcheck_service.ts]: port in host seems to be hardcoded here!

File: src/distributor/rmq_healthcheck_service.ts [hunk 2]
```diff
@@ -118,7 +125,13 @@ export default class RMQModeHeathCheck {
 
   private async getQueueMetrics(queue: string): Promise<RMQMetric> {
     try {
-      const url = `http://${process.env.RMQ_HOST}:15672/api/queues/%2F/${queue}`
+      const protocol = process.env.RMQ_PROTOCOL === 'amqps' ? 'https' : 'http'
+      let host = `${process.env.RMQ_HOST}`
+      if (protocol === 'http') {
+        host += ':15672'
```
Comment (devendra-shardeum) [inline src/distributor/rmq_healthcheck_service.ts]: and here as well ..

File: src/messaging/rabbitmq/conn.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export default class RMQConnection {
 
   public async getConnection(): Promise<Connection> {
     this.conn = await amqp.connect({
-      protocol: 'amqp',
+      protocol: process.env.RMQ_PROTOCOL || 'amqp',
       hostname: process.env.RMQ_HOST,
       port: process.env.RMQ_PORT ? parseInt(process.env.RMQ_PORT) : 5672,
```
Comment (devendra-shardeum) [inline src/messaging/rabbitmq/conn.ts:16]: default fallback should be 15672 and not 5672 .. no?

Comment (devendra-shardeum) [reply src/messaging/rabbitmq/conn.ts:16]: line no. 16

Comment (aniketdivekar) [reply src/messaging/rabbitmq/conn.ts:16]: 5672 is the port for connecting to broker for publishing message, 15672 is management API port

**Review by devendra-shardeum — APPROVED** [review]
code changes approved.. testing done on previous commit in this pr

**Pull request merged**